### ISSUE TITLE
Testing: Add lint rule to forbid `<a title>`

### ISSUE THREAD

**Issue by aduth** [WordPress/gutenberg#20198]
Related: #19990, #19735

Based on the following resources, it is my understanding that we should never want to include a `title` attribute on a link.

- https://core.trac.wordpress.org/ticket/24766
- https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute
- https://silktide.com/blog/2013/i-thought-title-text-improved-accessibility-i-was-wrong
- https://developer.paciellogroup.com/blog/2012/01/html5-accessibility-chops-title-attribute-use-and-abuse/
- https://www.24a11y.com/2017/the-trials-and-tribulations-of-the-title-attribute/
- https://a11yproject.com/posts/title-attributes/

Much like how we already use [`eslint-plugin-jsx-a11y`](https://github.com/evcohen/eslint-plugin-jsx-a11y) for linting various accessibility issues in JSX usage, we should be able to forbid the use of this attribute on links.

**Implementation Notes:**

It is unclear if this might be best implemented as:

- An upstream patch to [`eslint-plugin-jsx-a11y`](https://github.com/evcohen/eslint-plugin-jsx-a11y) which is then incorporated into this project.
- A project-specific rule, like many of the [existing `no-restricted-syntax`](https://github.com/WordPress/gutenberg/blob/bf150391900efe247aaede43b009c7019c76c837/.eslintrc.js#L45).
- A new rule in [`@wordpress/eslint-plugin`](https://github.com/WordPress/gutenberg/tree/master/packages/eslint-plugin), and whether or not that should be part of its bundled `recommended` ruleset.

Important note: `title` attributes can be expected ([_encouraged_](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/iframe-has-title.md) even) for some non-link elements, so the rule should be made specific to links.

**Recommendation:** An upstream patch could serve as useful validation of the idea. Even if rejected and still desirable for our own project, the implementation could be reused for a custom rule.

**Comment by im3dabasia** [comment]
Hi @aduth , I can take up this issue. Let me know if you think this issue is still relevant. 

cc: @t-hamano

**Comment by aduth** [comment]
@im3dabasia Sorry for the late response!

Revisiting this, the issue is still valid, but the landscape has changed a bit since then. These days we'd likely recommend developers use `@wordpress/ui`'s `Link` for creating links. If we were worried about `title` usage, we could probably leverage the central component to control this, such as runtime validation with debug logging. It's a tight balance since `@wordpress/ui` components are low level and generally unopinionated. At the same time, it's quite unlikely there's ever a valid case for using `title` on links.

Given that this hasn't been a widespread issue since 2020, I think we could close this and revisit if the issue reemerges. Another angle to this is making sure that the patterns that would draw people to using `title` have clearly documented alternatives in the design system. This will vary case-by-case, but [infotips](https://wordpress.github.io/gutenberg/?path=/docs/design-system-components-popover--docs#infotip) or [IconButton](https://wordpress.github.io/gutenberg/?path=/docs/design-system-components-iconbutton--docs) are a couple examples.

(cc @WordPress/gutenberg-components )
